tests/by-util/test_cp.rs

```diff
@@ -2660,6 +2660,71 @@ fn test_copy_through_dangling_symlink_no_dereference() {
         .no_stdout();
 }
 
+#[test]
+fn test_cp_symlink_overwrite_detection() {
+    let ts = TestScenario::new(util_name!());
+    let at = &ts.fixtures;
+
+    at.mkdir("good");
+    at.mkdir("tmp");
+    at.write("README", "file1");
+    at.write("good/README", "file2");
+
+    ts.ccmd("ln")
+        .arg("-s")
+        .arg("foo")
+        .arg("tmp/README")
+        .succeeds();
+
+    at.touch("tmp/foo");
+
+    ts.ucmd()
+        .arg("README")
+        .arg("good/README")
+        .arg("tmp")
+        .fails()
+        .stderr_only(if cfg!(target_os = "windows") {
+            "cp: will not copy 'good/README' through just-created symlink 'tmp\\README'\n"
+        } else if cfg!(target_os = "macos") {
+            "cp: will not overwrite just-created 'tmp/README' with 'good/README'\n"
```

Comment:
Symlinks on windows? I thought windows doesn't have symlinks?

Why doesn't the macos error message say the same thing? I'm confused.

Reply:
I believe newer windows have symlinks. https://blogs.windows.com/windowsdeveloper/2016/12/02/symlinks-windows-10/
About the macos error message, i have no clue why it doesn't print the same thing. The CI was complaining so i changed it (its arguably lazy coding). We really need access to a mac for testing the cp behaviour on there.